libgalois/include/katana/analytics/skip_gram/skip_gram.h

```diff
@@ -0,0 +1,101 @@
+#ifndef KATANA_LIBGALOIS_KATANA_ANALYTICS_SKIPGRAM_SKIPGRAM_H_
+#define KATANA_LIBGALOIS_KATANA_ANALYTICS_SKIPGRAM_SKIPGRAM_H_
+
+#include <fstream>
+#include <iostream>
+#include <map>
+
+#include "katana/AtomicHelpers.h"
+#include "katana/AtomicWrapper.h"
+#include "katana/PropertyGraph.h"
+#include "katana/analytics/Plan.h"
+
+// API
+
+namespace katana::analytics {
+
+/// A computational plan to for Skip Gram Embedding
+class SkipGramPlan : public Plan {
+public:
+  enum Algorithm {
+    kSkipGram,
+  };
+
+  static const uint32_t kEmbeddingSize = 100;
+  static constexpr double kAlpha = 0.025;
+  static const uint32_t kWindow = 5;
+  static constexpr double kDownSampleRate = 0.001;
+  static const bool kHierarchicalSoftmax = false;
+  static const uint32_t kNumNegSamples = 5;
+  static const uint32_t kNumIterations = 5;
+  static const uint32_t KMinimumFrequency = 5;
+
+private:
+  Algorithm algorithm_;
+  //Size of the embedding vector
+  uint32_t embedding_size_;
+  //alpha
+  double alpha_;
+  //window size
+  uint32_t window_;
+  //down-sampling rate
+  double down_sample_rate_;
+  //Enable/disable hierarchical softmax
+  bool hierarchical_softmax_;
+  //Number of negative samples
+  uint32_t num_neg_samples_;
+  //Number of Training Iterations
+  uint32_t num_iterations_;
+  //Mininum Frequency
+  uint32_t minimum_frequency_;
```

Comment:
The documentation should be on the accessors below so that the documentation is on the public API. And they should have `///` so they are doxygen doc comments.